src/main/java/algorithms/curated170/medium/maze2/TheMazeIIDijkstra.java

```diff
@@ -0,0 +1,77 @@
+package algorithms.curated170.medium.maze2;
+
+import java.util.Arrays;
+import java.util.PriorityQueue;
+
+ 
+public class TheMazeIIDijkstra {
+ 
+    int[][] maze;
+  public int shortestDistance(int[][] maze, int[] start, int[] dest) {
+ 
+        this.maze = maze;
+        
+        return dijkstra(start, dest); 
+    }
+   
+ public int dijkstra( int[] start, int[] destination) {
+         
+        PriorityQueue<int[]> pq = new PriorityQueue<>((p1, p2) -> p1[2] - p2[2]); 
+        int[][] directions = {{0, -1}, {1, 0}, {-1, 0}, {0, 1}};
+     
+        pq.add(new int[]{start[0], start[1], 0});
+        while (!pq.isEmpty()) {
+        
+            int[] startPoint = pq.poll();
+           
+            if (startPoint[0] < 0)
+                break;
```

Comment:
I don't understand the function of this statement here. Can you please explain that?